rids/monitors/ip_monitor.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2022 Jigsaw Operations LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""IP monitoring classes and functionality.
+
+IpPacketMonitor for obtaining remote IPs.
+IpPacket as the type for communicating to related rules.
+IpRule for representing the IOCs that packets should be checked against.
+"""
+
+from dataclasses import dataclass
+import ipaddress
+from typing import Union
+from typing import Generator
+
+from rids.monitors import tshark
+
+
+IpAddress = Union[ipaddress.IPv4Address, ipaddress.IPv6Address]
+
+
+@dataclass
+class IpPacket:
+  """Represents a remote IP endpoint and when it was seen."""
+  timestamp: str  # TODO: if calculating duration with this, make it a datetime
```

Comment:
This is not simply a string. It's a tshark timestamp string.
We are using a tshark type definition in our code that is supposed to be agnostic to tshark. This kind of hidden dependency makes it harder to replace the source of packets, for example. It makes the code more rigid (which is counter-intuitive).

It's fine for now, but using external types or time as string is a major source of pain. Please add a TODO to use datetime instead.

Same for `TlsConnection`.

Reply:
That's a valid concern -- it would make it a hurdle if migrating away from tshark to a different packet capture solution without normalizing the format being shared or logged.  Especially if we were depending on the time format already by then, but we could still normalize to the time format that tshark chose.  We're not mixing different timestamp formats, at least.  Anyway, I'm fine with changing it now too, it's not a big change to do up front.  We should agree on what is being output, though, and whether to use UTC or at least an ISO format that includes source time zone.  We'll have to trust the TZ indicators in the environment.

Reply:
done